molgenis-data/src/main/java/org/molgenis/data/meta/AttributeMetaDataRepositoryDecorator.java

```diff
@@ -364,37 +364,44 @@ private void validateUpdate(AttributeMetaData currentAttr, AttributeMetaData new
-		DATA_TYPE_ALLOWED_TRANSITIONS.put(DATE, EnumSet.of(DATE_TIME, HTML, SCRIPT, STRING, TEXT));
-		DATA_TYPE_ALLOWED_TRANSITIONS.put(DATE_TIME, EnumSet.of(DATE, HTML, SCRIPT, STRING, TEXT));
-		DATA_TYPE_ALLOWED_TRANSITIONS.put(DECIMAL, EnumSet.of(HTML, INT, LONG, SCRIPT, TEXT));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(DATE, EnumSet.of(CATEGORICAL, DATE_TIME, HTML, SCRIPT, STRING, TEXT, XREF));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(DATE_TIME, EnumSet.of(CATEGORICAL, DATE, HTML, SCRIPT, STRING, TEXT, XREF));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(DECIMAL, EnumSet.of(CATEGORICAL, HTML, INT, LONG, SCRIPT, TEXT, XREF));
 		DATA_TYPE_ALLOWED_TRANSITIONS.put(EMAIL, EnumSet.of(SCRIPT, HTML, STRING, TEXT));
 		DATA_TYPE_ALLOWED_TRANSITIONS.put(ENUM, EnumSet.noneOf(AttributeType.class));
 		DATA_TYPE_ALLOWED_TRANSITIONS.put(FILE, EnumSet.noneOf(AttributeType.class));
 		// not implemented: HTML -> CATEGORICAL, CATEGORICAL_MREF, EMAIL, ENUM, HYPERLINK, MREF, XREF
 		// not allowed    : HTML -> COMPOUND, FILE, HTML
-		DATA_TYPE_ALLOWED_TRANSITIONS
-				.put(HTML, EnumSet.of(BOOL, DATE, DATE_TIME, DECIMAL, INT, LONG, SCRIPT, STRING, TEXT));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(HTML,
+				EnumSet.of(CATEGORICAL, BOOL, DATE, DATE_TIME, DECIMAL, INT, LONG, SCRIPT, STRING, TEXT, XREF));
 		DATA_TYPE_ALLOWED_TRANSITIONS.put(HYPERLINK, EnumSet.of(SCRIPT, HTML, STRING, TEXT));
-		DATA_TYPE_ALLOWED_TRANSITIONS.put(INT, EnumSet.of(DECIMAL, HTML, LONG, SCRIPT, TEXT));
-		DATA_TYPE_ALLOWED_TRANSITIONS.put(LONG, EnumSet.of(DECIMAL, HTML, INT, SCRIPT, TEXT));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(INT, EnumSet.of(CATEGORICAL, DECIMAL, HTML, LONG, SCRIPT, TEXT, XREF));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(LONG, EnumSet.of(CATEGORICAL, DECIMAL, HTML, INT, SCRIPT, TEXT, XREF));
 		DATA_TYPE_ALLOWED_TRANSITIONS.put(MREF, EnumSet.of(CATEGORICAL_MREF));
 		// not implemented: SCRIPT -> CATEGORICAL, CATEGORICAL_MREF, EMAIL, ENUM, HYPERLINK, MREF, XREF
 		// not allowed    : SCRIPT -> COMPOUND, FILE, SCRIPT
-		DATA_TYPE_ALLOWED_TRANSITIONS
-				.put(SCRIPT, EnumSet.of(BOOL, DATE, DATE_TIME, DECIMAL, HTML, INT, LONG, STRING, TEXT));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(SCRIPT,
+				EnumSet.of(CATEGORICAL, BOOL, DATE, DATE_TIME, DECIMAL, HTML, INT, LONG, STRING, TEXT, XREF));
 		// not implemented: TEXT -> CATEGORICAL, CATEGORICAL_MREF, EMAIL, ENUM, HYPERLINK, MREF, XREF
 		// not allowed    : TEXT -> COMPOUND, FILE, TEXT
 		// not implemented: STRING -> CATEGORICAL, CATEGORICAL_MREF, EMAIL, ENUM, HYPERLINK, MREF, XREF
 		// not allowed    : STRING -> COMPOUND, FILE, STRING
-		DATA_TYPE_ALLOWED_TRANSITIONS
-				.put(STRING, EnumSet.of(BOOL, DATE, DATE_TIME, DECIMAL, HTML, INT, LONG, SCRIPT, TEXT));
-		DATA_TYPE_ALLOWED_TRANSITIONS
-				.put(TEXT, EnumSet.of(BOOL, DATE, DATE_TIME, DECIMAL, HTML, INT, LONG, SCRIPT, STRING));
-		DATA_TYPE_ALLOWED_TRANSITIONS.put(XREF, EnumSet.of(CATEGORICAL));
+		DATA_TYPE_ALLOWED_TRANSITIONS.put(STRING,
```

Comment:
TypeTest: When I want to change xstring type to an XREF I get a nullpointer:

java.lang.NullPointerException: null
	at org.molgenis.data.postgresql.PostgreSqlQueryGenerator.getPostgreSqlType(PostgreSqlQueryGenerator.java:811) ~[molgenis-data-postgresql-2.0.0-SNAPSHOT.jar:na]
	at org.molgenis.data.postgresql.PostgreSqlQueryGenerator.getSqlSetDataType(PostgreSqlQueryGenerator.java:90) ~[molgenis-data-postgresql-2.0.0-SNAPSHOT.jar:na]